src/components/templates/Auditorium/components/Section/Section.tsx

```diff
@@ -0,0 +1,153 @@
+
+export interface SectionProps {
+  venue: AuditoriumVenue;
+}
+
+export const Section: React.FC<SectionProps> = ({ venue }) => {
+  const {
+    iframeUrl,
+    rows = DEFAULT_ROWS_NUMBER,
+    columns = DEFAULT_COLUMNS_NUMBER,
+  } = venue;
+
+  const { sectionId } = useParams<{ sectionId: string }>();
+  const venueId = useVenueId();
+
+  const {
+    auditoriumSection,
+    getUserBySeat,
+    takeSeat,
+    leaveSeat,
+  } = useAuditoriumSection({
+    venueId,
+    sectionId,
+  });
+
+  // NOTE: Video takes 1/3 of the seats
+  const videoWidthInSeats = useMemo(
+    () => Math.max(Math.floor(columns / 3), VIDEO_MIN_WIDTH_IN_SEATS),
+    [columns]
+  );
+
+  // NOTE: Keep the 16:9 ratio
+  const videoHeightInSeats = useMemo(
+    () => Math.ceil(videoWidthInSeats * (9 / 16)),
+    [videoWidthInSeats]
+  );
+
+  const checkIfSeat = useCallback(
+    (rowIndex: number, columnIndex: number) => {
+      const translatedRowIndex = translateIndex({
+        index: rowIndex,
+        totalAmount: rows,
+      });
+      const translatedColumnIndex = translateIndex({
+        index: columnIndex,
+        totalAmount: columns,
+      });
+
+      const isInVideoRow =
+        Math.abs(translatedRowIndex) <= videoHeightInSeats / 2;
+      const isInVideoColumn =
+        Math.abs(translatedColumnIndex) <= videoWidthInSeats / 2;
+
+      return !(isInVideoRow && isInVideoColumn);
+    },
+    [rows, columns, videoHeightInSeats, videoWidthInSeats]
+  );
+
+  useEffect(() => {
+    return () => {
+      leaveSeat();
+    };
+  }, [leaveSeat]);
+
+  const iframeInlineStyles = useMemo(
+    () => ({
+      width: `calc(${videoWidthInSeats} * (${SEAT_SIZE} + ${SEAT_SPACING}))`,
+      height: `calc(${videoHeightInSeats} * (${SEAT_SIZE} + ${SEAT_SPACING}))`,
+      minWidth: `calc(${videoWidthInSeats} * (${SEAT_SIZE_MIN} + ${SEAT_SPACING}))`,
+      minHeight: `calc(${videoHeightInSeats} * (${SEAT_SIZE_MIN} + ${SEAT_SPACING}))`,
+    }),
+    [videoWidthInSeats, videoHeightInSeats]
+  );
+
+  if (!auditoriumSection) return <p>The section id is invalid</p>;
+
+  return (
+    <div className="section">
+      <div className="section__seats">
+        <Video
+          overlayClassname="section__video-overlay"
+          iframeClassname="section__video"
+          iframeStyles={iframeInlineStyles}
+          src={iframeUrl}
+        />
+        {Array.from(Array(DEFAULT_ROWS_NUMBER)).map((_, rowIndex) => (
+          <div key={rowIndex} className="section__seats-row">
+            {Array.from(Array(DEFAULT_COLUMNS_NUMBER)).map((_, columnIndex) => {
+              const user = getUserBySeat({
+                row: rowIndex,
+                column: columnIndex,
+              });
+
+              if (user) {
+                return (
+                  <UserProfilePicture
+                    user={user}
+                    avatarClassName={"section__user-avatar"}
+                    setSelectedUserProfile={() => {}}
+                  />
```

Comment:
@0xdevalias they will be important, when I add Reactions here, but I don't know what should we do with them. It seems to be a really nice feature, but poorly implemented performance-wise, code-wise, security-wise(@Sparkle-T had warns from a third-party plugin that there are security problems, when the reactins were turned on)

Reply:
I would say that we should continue to follow the pattern in our existing code, and then when we refactor to fix performance/etc (as we have planned to happen very soon, eg. this sprint) then we can apply those fixes consistently; rather than having to account for different custom situations/etc.

Until @malaikaaiyar's fix in https://github.com/sparkletown/sparkle/pull/1290 lands, not including the `isAudioEffectDisabled` prop  (set either 'properly' with state, or statically to `true`) here will result in the bug described in https://github.com/sparkletown/internal-sparkle-issues/issues/310

The warnings that @Sparkle-T mentioned in https://github.com/sparkletown/internal-sparkle-issues/issues/496 are a false positive in the 3rd party tool, and not something we need to be concerned about.

Reply:
I lean towards adding Reactions as a separate PR. Will have to Create Reactions component + move the logic into the appropriate hook/hooks. Did you intend to do it in your UserProfile refactor?

Reply:
> I lean towards adding Reactions as a separate PR.

I think that's ok/i'm fine with that. You'll still need to make sure to pass through that `isAudioEffectDisabled` prop though.

> Did you intend to do it in your UserProfile refactor?

Umm.. not 100% tbh.. I think that refactor probably wouldn't be creating the 'reactions controls' part of things that is shown in the centre of the auditorium/etc. That should be a fairly easy standalone PR for someone to do though. Basically just wrapping up the existing duplicated code into a new component.

Reply:
Added in  `isAudioEffectDisabled ` in https://github.com/sparkletown/sparkle/pull/1251/commits/4525887a96f74fe7e8a1e8316d75e0aa3946aac9

Created an issue for Reactions
https://github.com/sparkletown/internal-sparkle-issues/issues/637